### PR TITLE
Inject stylesheets as links, remove links moving, inject script by a …

### PR DESCRIPTION
**Pull request by oleh-momot**
…fragment`s hook

### CODE REVIEW

File: ilc/server/tailor/fragment-hooks.js [hunk 1]
```diff
@@ -18,11 +18,15 @@ function insertStart(stream, attributes, headers, index) {
...
-                    : `<link rel="stylesheet" href="${ref.uri}"${
-                        id !== undefined
-                            ? ` data-fragment-id="${id}"`
-                            : ''
-                        }>`
+                    : id
+                        ?
+                            '<script>(function(url, id){' + 
+                                `const link = document.head.querySelector('link[data-fragment-id="' + id + '"]');` +
+                                'if (link) {' +
```
Comment (StyleT) [inline ilc/server/tailor/fragment-hooks.js]: Could we also check that `href !== url` to potentially avoid extra DOM operation

File: ilc/server/tailor/fragment-hooks.js [hunk 2]
```diff
@@ -18,11 +18,15 @@ function insertStart(stream, attributes, headers, index) {
...
+                                    `link.href = url;` +
+                                '}' +
+                            `})("${ref.uri}", "${id}");document.currentScript.remove();</script>`
```
Comment (StyleT) [inline ilc/server/tailor/fragment-hooks.js]: `document.currentScript` && `ChildNode.remove()` doesn't work in IE :(

Comment (StyleT) [reply ilc/server/tailor/fragment-hooks.js]: I think we can keep this script w/o removal if you won't be able to find a good solution quickly.

Comment (oleh-momot) [reply ilc/server/tailor/fragment-hooks.js]: I forgot we load polyfills near the end of the body
I decided to remove it now

File: ilc/server/tailor/fragment-hooks.js (hunk 1 above)
Comment (oleh-momot) [reply ilc/server/tailor/fragment-hooks.js]: Yeah, I missed it
Added

File: ilc/server/tailor/fragment-hooks.js [hunk 3]
```diff
@@ -18,11 +18,15 @@ function insertStart(stream, attributes, headers, index) {
...
                     ? `<!-- Async fragments are not fully implemented yet: ${ref.uri} -->`
-                    : `<link rel="stylesheet" href="${ref.uri}"${
-                        id !== undefined
-                            ? ` data-fragment-id="${id}"`
-                            : ''
-                        }>`
+                    : id
+                        ?
+                            '<script>(function(url, id){' + 
+                                `const link = document.head.querySelector('link[data-fragment-id="' + id + '"]');` +
```
Comment (StyleT) [inline ilc/server/tailor/fragment-hooks.js]: const -> var

File: ilc/server/tailor/configs-injector.js [hunk 4]
```diff
@@ -155,6 +128,10 @@ module.exports = class ConfigsInjector {
         return `<link rel="preload" href="${url}" as="script" ${this.#getCrossoriginAttribute(url)}>`;
     };
 
+    #wrapWithFragmentStylesheetLink = (url, fragmentId) => {
+        return `<link rel="stylesheet" type="text/css" href="${url}" data-fragment-id="${fragmentId}">`;
```
Comment (StyleT) [inline ilc/server/tailor/configs-injector.js]:
```suggestion
        return `<link rel="stylesheet" href="${url}" data-fragment-id="${fragmentId}">`;
```

File: ilc/server/tailor/configs-injector.js [hunk 5]
```diff
@@ -24,53 +24,32 @@ module.exports = class ConfigsInjector {
...
         ) + '</head>');
 
-        const scriptRefs = await this.#getRouteScriptRefs(reqUrl);
-
+        const routeAssets = await this.#getRouteAssets(reqUrl);
+        
         document = document.replace('<head>', '<head>' + this.#wrapWithIgnoreDuringParsing(
-            ...scriptRefs.map(this.#wrapWithLinkToPreloadScript),
+            ...routeAssets.scriptLinks,
+            ...routeAssets.stylesheetLinks,
```
Comment (StyleT) [inline ilc/server/tailor/configs-injector.js]: Let's make stylesheet links to be higher then script preloads

File: ilc/server/tailor/configs-injector.js (hunk 4 above)
Comment (StyleT) [reply ilc/server/tailor/configs-injector.js]: It's not required with the HTML5 spec

Comment (oleh-momot) [reply ilc/server/tailor/configs-injector.js]: Removed

File: ilc/server/tailor/configs-injector.js (hunk 5 above)
Comment (oleh-momot) [reply ilc/server/tailor/configs-injector.js]: Moved

**Pull request merged**